### PR TITLE
feat: add duptable support for `reth db get`

### LINKED ISSUE #2836: Add duptable support for `reth db get`
### Describe the feature

`reth db get` does not currently support duptables but it would be nice if it did

### Additional context

https://github.com/paradigmxyz/reth/pull/2772#pullrequestreview-1436354459

### PR DESCRIPTION
**Pull request by dragan2234**
Closes #2836

### CODE REVIEW

**Comment by dragan2234** [comment]
Tables:
``` 
    (TableType::DupSort, PlainStorageState::const_name()),
    (TableType::DupSort, AccountChangeSet::const_name()),
    (TableType::DupSort, StorageChangeSet::const_name()),
    (TableType::DupSort, HashedStorage::const_name()),
    (TableType::DupSort, StoragesTrie::const_name()),
```

**Comment by dragan2234** [comment]
Right now it works this way:

`reth db get-dup <table> <key> <subkey>`

File: bin/reth/src/db/get.rs [hunk 1]
```diff
@@ -1,24 +1,39 @@
...
     pub table: String, // TODO: Convert to enum
 
     /// The key to get content for
     #[arg(value_parser = maybe_json_value_parser)]
     pub key: String,
+
 }
 
+#[derive(Parser, Debug)]
+pub struct DupCommand {
```
Comment (gakonst) [inline bin/reth/src/db/get.rs:22]: We want to keep it under get, no need for a new command

File: bin/reth/src/utils.rs [hunk 2]
```diff
@@ -90,6 +91,11 @@ impl<'a, DB: Database> DbTool<'a, DB> {
         self.db.view(|tx| tx.get::<T>(key))?.map_err(|e| eyre::eyre!(e))
     }
 
+    /// Grabs the content of the table for the given key and subkey
+    pub fn get_dup<T: DupSort>(&mut self, key: T::Key, subkey: T::SubKey) -> Result<Option<T::Value>> {
+        self.db.view(|tx| tx.cursor_dup_read::<T>().unwrap().seek_by_key_subkey(key, subkey)).unwrap().map_err(|e| eyre::eyre!(e))
```
Comment (gakonst) [inline bin/reth/src/utils.rs:96]: @rkrasiuk is this the right way? IIRC we cannot do this fully generically and need to use the structs we've defined for the subkeys by table?

File: bin/reth/src/db/get.rs [hunk 3]
```diff
@@ -81,6 +96,59 @@ impl Command {
...
+        dupsort_get!([
+            PlainStorageState,
+            AccountChangeSet,
+            StorageChangeSet,
+            HashedStorage,
+            StoragesTrie
+        ]);
+    }
+
+    fn table_key<T: Table>(&self) -> Result<T::Key, eyre::Error>
```
Comment (gakonst) [inline bin/reth/src/db/get.rs:134]: This looks redundant if you combine the two commands

**Review by gakonst — CHANGES_REQUESTED** [review]

**Comment by dragan2234** [comment]
Thanks for the review. Just to note: there is this pr: https://github.com/paradigmxyz/reth/pull/2935 which needs to be merged before merging this one since it changes a few things related to this pr

File: bin/reth/src/utils.rs (hunk 2 above)
Comment (dragan2234) [reply bin/reth/src/utils.rs:96]: pinging @rkrasiuk I'm back on the issue. `seek_by_key_subkey` and `cursor_dup_read` forces me to pass the Subkey here, so it's impossible having the same macro for both types of tables (`Table` and `DupSort`), even with new changes from https://github.com/paradigmxyz/reth/pull/2935

So if this call with `tx.cursor_dup_read::<T>().unwrap().seek_by_key_subkey(key, subkey)` is the correct way of pulling data, we would need to change macro from https://github.com/paradigmxyz/reth/pull/2935 and create 2 separate macros for "normal" and "DupSort" tables. OR we create a generic method for DupSort tables but keep the `view` function only having `fn view<T: Table>` without DupSort.


Let me know your thoughts.

**Comment by dragan2234** [comment]
what I'm looking at right now is how to generate the view_dupsort function inside the same macro, but only for DupSort table type. Since that function would need to have type view_dupsort<DupSor>. I'm looking into the macro's identifiers, and expressions but still struggling to do it correctly

**Pull request closed without merging**